engine/src/object.cpp

```diff
@@ -5361,6 +5349,20 @@ void MCObject::sync_mfocus(void)
         MCmousestackptr->getcard()->mfocus(MCmousex, MCmousey);
 }
 
+bool MCObject::isancestorof(MCObject *p_object)
+{
+    if (p_object == nil)
+        return false;
```

Comment:
I'm not sure - it would be possible to call it on something like `MCtemplatebutton` whose `parent` is `nil`, although I think in all the existing cases it is protected from that by checking the `opened` variable, which presumably is always false for `MCtemplate...`, but I don't know off the top of my head if it's generally true that you can't be `opened` without a `parent`. @runrevmark?